qiskit/pulse/ir/ir.py

```diff
@@ -16,283 +16,213 @@
-            instruction: the Pulse `Instruction` represented by this IR instruction.
-            initial_time (Optional): Starting time of the instruction. Defaults to ``None``
-        """
-        self._instruction = instruction
-        if initial_time is None:
-            self._initial_time = None
-        else:
-            self.initial_time = initial_time
+        self._time_table = defaultdict(lambda: None)
+        self._sequence = rx.PyDAG(multigraph=False)
+        self._sequence.add_nodes_from([InNode, OutNode])
 
     @property
-    def instruction(self) -> Instruction:
-        """Return the instruction associated with the IrInstruction"""
-        return self._instruction
+    def alignment(self) -> Alignment:
+        """Return the alignment of the SequenceIR"""
+        return self._alignment
 
     @property
-    def initial_time(self) -> int | None:
-        """A clock time (in terms of system ``dt``) when this instruction is issued.
-
-        .. note::
-            initial_time value defaults to ``None`` and can only be set to non-negative integer.
-        """
-        return self._initial_time
+    def inst_targets(self) -> set:
+        """Recursively return a set of all Instruction.inst_target in the SequenceIR"""
+        inst_targets = set()
+        for elm in self.elements():
+            if isinstance(elm, SequenceIR):
+                inst_targets |= elm.inst_targets
+            else:
+                inst_targets.add(elm.inst_target)
+        return inst_targets
 
     @property
-    def duration(self) -> int:
-        """The duration of the instruction (in terms of system ``dt``)."""
-        return self._instruction.duration
+    def sequence(self) -> PyDAG:
+        """Return the DAG sequence of the SequenceIR"""
+        return self._sequence
 
-    @initial_time.setter
-    def initial_time(self, value: int):
-        """Set ``initial_time``
+    def append(self, element: SequenceIR | Instruction) -> int:
+        """Append element to the SequenceIR
 
-        Args:
-            value: The integer value of ``initial_time``.
-
-        Raises:
-            PulseError: if ``value`` is not ``None`` and not non-negative integer.
+        Returns: The index of the added element in the sequence.
         """
-        if not isinstance(value, (int, np.integer)) or value < 0:
-            raise PulseError("initial_time must be a non-negative integer")
-        self._initial_time = value
-
-    def shift_initial_time(self, value: int) -> None:
-        """Shift ``initial_time``
+        return self._sequence.add_node(element)
 
-        Shifts ``initial_time`` to ``initial_time+value``.
+    def elements(self) -> list[SequenceIR | Instruction]:
+        """Return a list of all elements in the SequenceIR"""
+        return self._sequence.nodes()[2:]
 
-        Args:
-            value: The integer value by which ``initial_time`` is to be shifted.
+    def scheduled_elements(
+        self, recursive: bool = False
```

Comment:
Do we want to sort by the first element (or such option)?

Reply:
Yes, makes sense.